tests/directory_services/test_freeipa_basic.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+import pytest
+from middlewared.test.integration.assets.directory_service import (
+    directoryservice,
+    FREEIPA_IP,
+    FREEIPA_BASEDN,
+    FREEIPA_BINDDN,
+    FREEIPA_BINDPW,
+    FREEIPA_HOSTNAME,
+)
+from middlewared.test.integration.utils import call, ssh
+
+
+@pytest.fixture(scope="module")
+def do_freeipa_connection():
+    # Confirm DNS forward
+    res = ssh(f"host {FREEIPA_HOSTNAME}", complete_response=True)
+    # stdout: "<FREEIPA_HOSTNAME> has address <FREEIPA_IP>"
+    assert res['stdout'].split()[-1] == FREEIPA_IP
+
+    # DNS reverse
+    res = ssh(f"host {FREEIPA_IP}", complete_response=True)
+    # stdout: <FREEIPA_IP_reverse_format>.in-addr.arpa domain name pointer <FREEIPA_HOSTNAME>.
+    assert res['stdout'].split()[-1] == FREEIPA_HOSTNAME + "."
+
+    with directoryservice(
+        'LDAP',
+        credential={
+            'credential_type': 'LDAP_PLAIN',
+            'binddn': FREEIPA_BINDDN,
+            'bindpw': FREEIPA_BINDPW,
+        },
+        configuration={
+            'server_urls': [f'ldaps://{FREEIPA_IP}'],
+            'basedn': FREEIPA_BASEDN,
+            'schema': 'RFC2307BIS',
+            'search_bases': {
+                'base_user': 'cn=users,cn=accounts,dc=tn,dc=ixsystems,dc=net',
+                'base_group': 'cn=groups,cn=accounts,dc=tn,dc=ixsystems,dc=net',
+                'base_netgroup': 'cn=ng,cn=compat,dc=tn,dc=ixsystems,dc=net'
+            },
+            'validate_certificates': False
+        },
+        retrieve_user=False,
+    ) as ldap_conn:
+        yield ldap_conn
+
+
+def test_setup_and_enabling_freeipa(do_freeipa_connection):
+    ds = call('directoryservices.status')
+    assert ds['type'] == 'LDAP'
+    assert ds['status'] == 'HEALTHY'
+
+
+def test_verify_config():
```

Comment:
To run this by itself it needs to be:
```suggestion
def test_verify_config(do_freeipa_connection):
```
However, this and the previous test could, and maybe should, be merged into one test. 🤷‍♂️